src/main/java/com/microsoft/spring/data/gremlin/repository/support/SimpleGremlinRepository.java

```diff
@@ -54,13 +55,19 @@ public GremlinEntityInformation getGremlinEntityInformation() {
     }
 
     @Override
-    public List<T> findAllById(Iterable<ID> ids) {
-        throw new NotImplementedException("findAllById of Repository is not implemented yet");
+    public List<T> findAllById(@NonNull Iterable<ID> ids) {
+        final List<T> results = new ArrayList<>();
+
+        ids.forEach(id -> this.findById(id).ifPresent(results::add));
```

Comment:
Is it possible that findAll could be paralleled to improve the performance if we have lots of ids to find?

Reply:
yes, we can make that paralleled. But as a prototype, we prefer to consider performance in future.